cookiecutter/main.py

```diff
@@ -52,7 +52,8 @@ def expand_abbreviations(template, config_dict):
     return template
 
 
-def cookiecutter(template, checkout=None, no_input=False, extra_context=None):
+def cookiecutter(template, checkout=None, no_input=False,
```

Comment:
The reason I inserted it in the middle rather than at the end is that in ``cookiecutter/cli.py``, which is the only place this function is called, this function is invoked in this way - ``cookiecutter(template, checkout, no_input)``. This implies ``extra_context`` is an optional argument. But the flag ``overwrite_if_exists`` is always passed to all the functions that require it and putting it at the end will require code change like one of these 2 ways, which don't look that good imho.
    
    # first way
    extra_context = None
    cookiecutter(template, checkout, no_input, extra_context, overwrite_if_exists)

    # second way
    cookiecutter(template, checkout, no_input, overwrite_if_exists=overwrite_if_exists)

Please suggest better ways to deal with this, if any.